jumpscale/packages/tfgrid_solutions/chats/extend_kube.py

```diff
@@ -0,0 +1,120 @@
+        nodes_pools_zip = list(zip(nodes, pools))
+        selected_nodes = list(filter(lambda x: x[0].node_id not in old_nodes, nodes_pools_zip))
+        if len(selected_nodes) < self.nodes_count:
+            self.stop(
+                f"Failed to find resources to deploy {self.nodes_count}, available nodes are: {len(selected_nodes)}"
+            )
+        new_nodes = selected_nodes[: self.nodes_count]
+        network_view = deployer.get_network_view(workload.network_id)
+        master_ip = workload.ipaddress
+
+        self.reservations = []
+        for node, pool_id in new_nodes:
+            res = deployer.add_network_node(workload.network_id, node, pool_id)
+            if res:
+                for wid in res["ids"]:
+                    success = deployer.wait_workload(wid, breaking_node_id=node.node_id)
+                    if not success:
+                        raise StopChatFlow(f"Failed to add node {node.node_id} to network {wid}")
+            network_view = network_view.copy()
+            ip_address = network_view.get_free_ip(node)
+            if not ip_address:
+                raise StopChatFlow(f"No free IPs for network {network_name} on the specifed node" f" {node_id}")
+
+            self.md_show_update(f"Deploying worker on node {node.node_id}")
+            # Add public ip
+            public_id_wid = 0
+            if self.enable_public_ip:
+                public_id_wid, _ = deployer.create_public_ip(
+                    pool_id, node.node_id, solution_uuid=metadata.get("solution_uuid")
+                )
+
+            self.reservations.append(
+                deployer.deploy_kubernetes_worker(
+                    pool_id,
+                    node.node_id,
+                    workload.network_id,
+                    workload.cluster_secret,
+                    workload.ssh_keys,
+                    ip_address,
+                    master_ip,
+                    size=self.cluster_size,
+                    identity_name=None,
+                    description="",
+                    public_ip_wid=public_id_wid,
+                    **metadata,
+                )
+            )
+
+        self.success_workload_count = 0
+        zos = j.sals.zos.get()
+        for resv in self.reservations:
+            try:
+                success = deployer.wait_workload(resv, self, cancel_by_uuid=False)
+                self.success_workload_count += 1
+            except DeploymentFailed as ex:
+                # Cleaning k8s workloads and public IP workloads in case of failure in deployment
+                workload = zos.workloads.get(resv)
+                if hasattr(workload, "public_ip") and workload.public_ip:
+                    zos.workloads.decomission(workload.public_ip)
+                zos.workloads.decomission(wid)
+                j.logger.error(f"Failed to deploy  workloads for {resv}, the error: {str(ex)}")
+        if not self.success_workload_count:
```

Comment:
should add a check `if self.success_workload_count < len(self.reservations)` in that case it has partially failed